test-integration/DepositZap.js

```diff
@@ -0,0 +1,347 @@
+      bytes32("erc20Allocation"),
+      erc20AllocationAddress
+    );
+
+    const tvlManagerAddress = await generateContractAddress(deployer);
+    await addressRegistry.registerAddress(
+      bytes32("tvlManager"),
+      tvlManagerAddress
+    );
+
+    const lpAccountAddress = await generateContractAddress(deployer);
+    await addressRegistry.registerAddress(
+      bytes32("lpAccount"),
+      lpAccountAddress
+    );
+
+    await addressRegistry.registerAddress(
+      bytes32("adminSafe"),
+      adminSafe.address
+    );
+
+    await addressRegistry.registerAddress(
+      bytes32("emergencySafe"),
+      emergencySafe.address
+    );
+
+    const lpSafeAddress = await generateContractAddress(deployer);
+    await addressRegistry.registerAddress(bytes32("lpSafe"), lpSafeAddress);
+  });
+
+  before("Deploy mAPT", async () => {
+    const MetaPoolToken = await ethers.getContractFactory("TestMetaPoolToken");
+    const mAptLogic = await MetaPoolToken.deploy();
+    await mAptLogic.deployed();
+
+    const mAptInitData = MetaPoolToken.interface.encodeFunctionData(
+      "initialize(address)",
+      [addressRegistry.address]
+    );
+    const mAptProxy = await deployProxy(mAptLogic, proxyAdmin, mAptInitData);
+    mApt = await MetaPoolToken.attach(mAptProxy.address);
+
+    await addressRegistry.registerAddress(bytes32("mApt"), mApt.address);
+  });
+
+  before("Attack to Mainnet ERC20s", async () => {
+    curve3Crv = await ethers.getContractAt(
+      "IDetailedERC20",
+      CURVE_3CRV_ADDRESS
+    );
+    dai = await ethers.getContractAt("IDetailedERC20", DAI_ADDRESS);
+    usdc = await ethers.getContractAt("IDetailedERC20", USDC_ADDRESS);
+    tether = await ethers.getContractAt("IDetailedERC20", TETHER_ADDRESS);
+  });
+
+  before("Deploy Oracle Adapter", async () => {
+    const OracleAdapter = await ethers.getContractFactory("OracleAdapter");
+    oracleAdapter = await OracleAdapter.deploy(
+      addressRegistry.address,
+      tvlAgg.address,
+      [CURVE_3CRV_ADDRESS],
+      [USDC_AGG_ADDRESS],
+      86400,
+      86400
+    );
+    await oracleAdapter.deployed();
+    await addressRegistry.registerAddress(
+      bytes32("oracleAdapter"),
+      oracleAdapter.address
+    );
+  });
+
+  before("Deploy Index Token", async () => {
+    const IndexToken = await ethers.getContractFactory("TestIndexToken");
+    const logic = await IndexToken.deploy();
+    await logic.deployed();
+
+    const initData = IndexToken.interface.encodeFunctionData(
+      "initialize(address,address)",
+      [addressRegistry.address, curve3Crv.address]
+    );
+    const proxy = await deployProxy(logic, proxyAdmin, initData);
+    indexToken = await IndexToken.attach(proxy.address);
+  });
+
+  before("Deploy Deposit Zap", async () => {
+    const DepositZap = await ethers.getContractFactory("DepositZap");
+    depositZap = await DepositZap.deploy(indexToken.address);
+    await depositZap.deployed();
+  });
+
+  describe("deposit", () => {
```

Comment:
When the `minAmount` for deposit can be set instead of hardcoded zero, we should create a test to ensure it reverts and succeeds appropriately.